packages/shared/src/components/profile/TagList.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactElement } from 'react';
+import { UserShortInfoPlaceholder } from './UserShortInfoPlaceholder';
+import InfiniteScrolling, {
+  InfiniteScrollingProps,
+} from '../containers/InfiniteScrolling';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../typography/Typography';
+import {
+  ContentPreference,
+  ContentPreferenceType,
+} from '../../graphql/contentPreference';
+import { FollowButton } from '../contentPreference/FollowButton';
+import { useFeedSettingsEditContext } from '../feeds/FeedSettings/FeedSettingsEditContext';
+import { CopyType } from '../sources/SourceActions/SourceActionsFollow';
+
+export interface TagListProps {
+  scrollingProps: Omit<InfiniteScrollingProps, 'children'>;
+  tags: ContentPreference[];
+  placeholderAmount?: number;
+  isLoading?: boolean;
+  emptyPlaceholder?: JSX.Element;
+}
+
+export const TagList = ({
```

Comment:
This seemed neater to use as it can handle paginated loading and directly uses the contentPreference vibe.